portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -99,4 +125,79 @@ public void storeMarker(Marker marker) {
 
     datastore.put(markerEntity);
   }
+
+  public boolean repeatMarkers(double reportLat, double reportLng, String reportDate,
+   String reportTime, String reportCrime) {
+    Query query = new Query(ENTITY_TITLE);
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String crime = (String) entity.getProperty("crimeType");
+      String date = (String) entity.getProperty("date");
+      String time = (String) entity.getProperty("time");
+
+      if (sameLocation(lat, lng, reportLat, reportLng)) 
```

Comment:
instead of nested if conditions better to have a single if condition like:
IF (condition1 AND condition2 AND condition3)

Reply:
Changed!